BaseClasses.py

```diff
@@ -7,10 +7,10 @@
 import secrets
 from argparse import Namespace
 from collections import Counter, deque
-from collections.abc import Collection, MutableSequence
+from collections.abc import MutableSequence
 from enum import IntEnum, IntFlag
-from typing import (AbstractSet, Any, Callable, ClassVar, Dict, Iterable, Iterator, List, Mapping, NamedTuple,
-                    Optional, Protocol, Set, Tuple, Union, TYPE_CHECKING)
+from typing import (AbstractSet, Any, Callable, ClassVar, Dict, Generic, Iterable, Iterator, List, Mapping, NamedTuple,
+                    Optional, Protocol, Set, Tuple, TypeVar, Union, TYPE_CHECKING)
 
 from typing_extensions import NotRequired, TypedDict
```

Comment:
```suggestion
from typing import (AbstractSet, Any, Callable, ClassVar, Dict, Generic, Iterable, Iterator, List, Mapping, NamedTuple,
                    Optional, Protocol, Set, Tuple, Union, TYPE_CHECKING)

from typing_extensions import NotRequired, TypedDict, TypeVar
```
so that you can set a default type